otwcode/otwarchive
# AO3-6161 Only allow jpg/gif/png as pseud icons.

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6161

## Purpose
When editing pseud icons, only icons with the mime types for GIF, JPEG, and PNG validate.
Icons with other mime types are rejected, so users will not be able to create new pseuds or update existing ones with icons using other mime types.

I still have some open questions:
* Is there a planned migration for icons which are currently in "invalid" formats? Currently, users could not make changes to such pseuds unless the icons are "fixed" too.
* Is the default error message of "Icon content type is invalid" readable enough? (This came for free - anything more complicated would need actual i18n work)
* Is the allowlist approach (only allowing exactly the mime types for GIF, JPEG and PNG) too restrictive? I chose it because the denylist approach did not match the description on the "Edit Pseud" page (and also because .bmp files can have weirder mime types to check)

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?
* Select or create a user.
    * Create a new pseud, selecting a .bmp file (should fail) or a .jpg/.gif/.png file (should succeed) as an icon.
    * Edit a current pseud, selecting a .bmp file (should fail) or a .jpg/.gif/.png file (should succeed) as an icon.
    * Edit a current pseud, select "Delete your icon and revert to our default" and save. This should still succeed (the default icon is a .png)

## Credit
JIRA account: https://otwarchive.atlassian.net/jira/people/557058%3A0ed2d036-cfdd-4793-9df9-3ec3e141a3cf
Please credit me as "Rudolf M. Schreier (ThePadawan)" using "he/she/they" pronouns.
